src/waitFor.ts

```diff
@@ -177,7 +177,10 @@ function waitForInternal<T>(
         }
       }
       if (typeof onTimeout === 'function') {
-        onTimeout(error);
+        const result = onTimeout(error);
+        if (result) {
+          error = result;
```

Comment:
I could probably use a more thorough eye on this change. When a custom `onTimeout` was passed to `waitFor`, we did not previously allow for this to alter the returned error. This change makes it so the error returned by the custom `onTimeout` is the error that is ultimately reported.

Reply:
Resolved in 2a9fb5c. The behavior should now match DOM Testing Library where if a custom `onTimeout` is passed, that function receives the error without the element tree. The element tree is only added in the default `onTimeout` if no custom one is passed.